src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -61,622 +54,461 @@ const defaultProps = {
-        if (!this.isWebOrDesktop) {
-            return;
-        }
-        document.removeEventListener('focusin', this.handleFocusIn);
-        document.removeEventListener('focusout', this.handleFocusOut);
-    }
-
-    subscribeToEnterShortcut() {
-        const enterConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
-        this.unsubscribeEnter = KeyboardShortcut.subscribe(
-            enterConfig.shortcutKey,
-            this.selectFocusedOption,
-            enterConfig.descriptionKey,
-            enterConfig.modifiers,
-            true,
-            () => !this.state.allOptions[this.state.focusedIndex],
-        );
-    }
-
-    subscribeToCtrlEnterShortcut() {
-        const CTRLEnterConfig = CONST.KEYBOARD_SHORTCUTS.CTRL_ENTER;
-        this.unsubscribeCTRLEnter = KeyboardShortcut.subscribe(
-            CTRLEnterConfig.shortcutKey,
-            () => {
-                if (this.props.canSelectMultipleOptions) {
-                    this.props.onConfirmSelection();
-                    return;
+    const selectRow = useCallback(
+        (option, ref) =>
+            new Promise((resolve) => {
+                if (props.shouldShowTextInput && props.shouldPreventDefaultFocusOnSelectRow) {
+                    if (relatedTarget.current && ref === relatedTarget.current) {
+                        textInputRef.current.focus();
+                        relatedTarget.current = null;
+                    }
+                    if (textInputRef.current.isFocused()) {
+                        setSelection(textInputRef.current, 0, value.length);
+                    }
                 }
+                const selectedOption = props.onSelectRow(option);
+                resolve(selectedOption);
 
-                const focusedOption = this.state.allOptions[this.state.focusedIndex];
-                if (!focusedOption) {
+                if (!props.canSelectMultipleOptions) {
                     return;
                 }
 
-                this.selectRow(focusedOption);
-            },
-            CTRLEnterConfig.descriptionKey,
-            CTRLEnterConfig.modifiers,
-            true,
-        );
-    }
-
-    unSubscribeFromKeyboardShortcut() {
-        if (this.unsubscribeEnter) {
-            this.unsubscribeEnter();
-        }
+                // Focus the first unselected item from the list (i.e: the best result according to the current search term)
+                setFocusedIndex(props.selectedOptions.length);
+            }),
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [props.shouldShowTextInput, props.shouldPreventDefaultFocusOnSelectRow, value.length, props.canSelectMultipleOptions, props.selectedOptions.length],
+    );
 
-        if (this.unsubscribeCTRLEnter) {
-            this.unsubscribeCTRLEnter();
-        }
-    }
+    const selectFocusedOption = useCallback(
+        (e) => {
+            const focusedItemKey = lodashGet(e, ['target', 'attributes', 'id', 'value']);
```

Comment:
Add some default value.
```suggestion
            const focusedItemKey = lodashGet(e, ['target', 'attributes', 'id', 'value'], '');
```